go/pkg/cas/upload.go

```diff
@@ -64,8 +65,7 @@ func (c *Client) Upload(ctx context.Context, inputC <-chan *UploadInput) (stats
 		Client: c,
 		eg:     eg,
 
-		fsCache: map[string]*fsCacheValue{},
-		fsSem:   semaphore.NewWeighted(int64(c.FSConcurrency)),
+		fsSem: semaphore.NewWeighted(int64(c.FSConcurrency)),
```

Comment:
initialize `fsCache: cache.getNSCache("filemetadata")`?

Reply:
Do you mean a process-global cache? This would mean that if new files were written and Upload is called again, those files might not be uploaded.

Reply:
Ah, I see. For our usage, the SDK code isn't long running - there's generally the assumption that files don't change mid build, so we do need some way of at least passing in a `fsCache` (and `Upload` can create a new one if none is provided?)

Reply:
I do think that we need `(absPath, mTime, size) => digest` cache, but it will come later.

Reply:
FYI the file metadata cache used to support invalidation, but we removed it due to the assumption Rubens mentioned.

https://github.com/bazelbuild/remote-apis-sdks/commit/6e74127154b6047b16f708f06ccfc46bb2a4172b#diff-a297d8d639b8a097cebe69978c28397a77fde5b4a4b6887205d0a12858bb2e8c